build.sh

```diff
@@ -1,8 +1,89 @@
 #!/usr/bin/env bash
 
+set -euo pipefail
+IFS=$'\n\t'
+
+usage() {
+    echo "usage: $0 [options]"
+    echo "options:"
+    echo "  --with-packages <dir>              use the specified archive of source-built packages to build."
+    echo "  --with-sdk <dir>                   use the specified SDK to build."
+}
+
+function abspath() {
+  # $1 : relative filename
+  parentdir=$(dirname "$1")
+
+  if [ -d "$1" ]; then
+      echo "$(cd "$1" && pwd)"
```

Comment:
(Nit) Inconsistent tabbing here, and the 2 space here vs. the rest of the file.

Reply:
Sorry. Opened https://github.com/dotnet/source-build-reference-packages/pull/102 for this.